### ISSUE TITLE
Question: Cosmos Db Consistency

### ISSUE THREAD

**Issue by mpaul31** [jet/equinox#212]
Hello,

I am reading through the documentation and trying to understand how this are setup and work with Cosmos Db. First, which consistency model must be used? Is multi readers / writers supported? I'm reading through the section **Reading from the Storage Model** and trying to work out in my head how we can be sure we are reading the latest state when `IfNoneMatch ` is `Found`  in the presence of multiple writers or not currently in cache. Isn't it possible we could be returned a document that is not the most recent and our command processing logic may in turn not emit emits thus leaving the system in an inconsistent state?

Thank you and any information you could provide would be great!

**Comment by bartelink** [comment]
TL;DR Ultimately every write is subject to an etag or expectedVersion matching in the write (in the sync stored proc)

The writing loop / consistency check scheme means it doesnt matter if the initial sync of the `Stream.Transact` fails to get the latest version, it will all get reconciled (even if it guaranteed latest, it can still be superseded). Any starting point will do (see xmldoc re `AllowStale`)

related: Equinox uses session consistency - there's some missing wiring wrt that :- https://github.com/jet/equinox/pull/195 - this would provide am full read-your-writes guarantee.

I'm working on diagrams atm that'll likely be relevant for you: https://github.com/jet/equinox/blob/diag/DOCUMENTATION.md#code-diagrams-for-equinoxcosmos
